### ISSUE TITLE
`org.logstash.ObjectMappers.RubyStringSerializer` Should Serialize a RubyString without creating a Java `String`

### ISSUE THREAD

**Issue by original-brownbear** [elastic/logstash#8110]
The current serializer for `RubyString` is ok but could be a lot faster.
We have:

```java
    /**
     * Serializer for {@link RubyString} since Jackson can't handle that type natively, so we
     * simply serialize it as if it were a {@link String}.
     */
    private static final class RubyStringSerializer
        extends NonTypedScalarSerializerBase<RubyString> {

        RubyStringSerializer() {
            super(RubyString.class, true);
        }

        @Override
        public void serialize(final RubyString value, final JsonGenerator generator,
            final SerializerProvider provider)
            throws IOException {
            generator.writeString(value.asJavaString());
        }
    }
```

which means we create a new `value.asJavaString()` for each `RubyString` we serialize.
This is unnecessary, we already have all the `byte[]` in the `ByteList` backing the `RubyString` and should simply write them to the output Stream.
This isn't so pressing since the `String` is actually never created and JDK8 escapes it to a `char[]` but still, allocating that is just wasted memory and CPU cycles.

**Comment by original-brownbear** [comment]
Once https://github.com/elastic/logstash/pull/8205 is merged, we can do this in a very fast way by writing a typed `RubyString` to CBOR as `byte[]` without any allocations necessary whatsoever :)

**Comment by original-brownbear** [comment]
so fix to this is already in #7950 :)

**Comment by original-brownbear** [comment]
Fix was merged with the Java execution (also works with Ruby exec since it's just a change to the String serializer).
